contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -5349,6 +5363,104 @@ pg_class_update_acl(Oid newViewOid, Acl *oldViewAcl)
 	table_close(pg_class_rel, RowExclusiveLock);
 }
 
+static List *
+get_old_view_column_acls(Oid oldViewOid)
+{
+	Relation pg_attribute_rel;
+	SysScanDesc scan;
+	ScanKeyData skey;
+	HeapTuple tuple;
+	List *column_acls = NIL;
+	
+	pg_attribute_rel = table_open(AttributeRelationId, AccessShareLock);
+	
+	ScanKeyInit(&skey,
+				Anum_pg_attribute_attrelid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(oldViewOid));
+	
+	scan = systable_beginscan(pg_attribute_rel, AttributeRelidNumIndexId,
+							true, NULL, 1, &skey);
+	
+	while (HeapTupleIsValid(tuple = systable_getnext(scan)))
+	{
+		Form_pg_attribute att = (Form_pg_attribute) GETSTRUCT(tuple);
+		Datum aclDatum;
+		bool isNull;
+		ColumnAclInfo *info;
+		
+		/* Skip system columns */
+		if (att->attnum <= 0)
+			continue;
+			
+		aclDatum = SysCacheGetAttr(ATTNAME, tuple, 
+								Anum_pg_attribute_attacl, &isNull);
+		
+		info = palloc(sizeof(ColumnAclInfo));
+		info->attnum = att->attnum;
+		info->attname = pstrdup(NameStr(att->attname));
+		info->acl = isNull ? NULL : DatumGetAclPCopy(aclDatum);
+		
+		column_acls = lappend(column_acls, info);
+	}
+	
+	systable_endscan(scan);
+	table_close(pg_attribute_rel, AccessShareLock);
+	
+	return column_acls;
+}
+
+static void
+restore_view_column_acls(Oid newViewOid, List *column_acls)
+{
+	ListCell *lc;
+	
+	foreach(lc, column_acls)
+	{
+		ColumnAclInfo *info = (ColumnAclInfo *) lfirst(lc);
+		Relation pg_attribute_rel;
+		HeapTuple attTup;
+		AttrNumber attnum = InvalidAttrNumber;
+		
+		/* Find the corresponding column in the new view by name */
+		attnum = get_attnum(newViewOid, info->attname);
```

Comment:
You already have info->att_num, why do you have to get attnum again from another api ?